lib_test/test_qcheck.ml

```diff
@@ -26,11 +27,18 @@ let arb_bencode =
             2, base;
           ]) n
   in
+  let rec shrink_int64 : int64 Q.Shrink.t =
+    fun x ->
+      let open Q.Iter in
+      ( ( return @@ Int64.div x 2_L )
+        <+> (return @@ Int64.(sub x one))
```

Comment:
here I think you need something like

```
if x > 0L then return (sub x one) else if x < 0L then return (plus x one) else nil
```

to ensure that the value is "decreasing" towards 0.

Or start with `if x = 0L then nil else …` to avoid infinite shrinking loops.

Reply:
also, don't do the recursion yourself, qcheck will call `shrink` again if the smaller value still triggers a failure? :-)